src/ajax.js

```diff
@@ -557,8 +557,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Uppercase the type
+		s.type = s.type.toUpperCase();
+
+		// Determine if request has content
+		s.hasContent = !rnoContent.test( s.type );
+
 		// Convert data if not already a string
-		if ( s.data && s.processData && typeof s.data !== "string" ) {
+		if ( s.data && ( s.processData || !s.hasContent ) && typeof s.data !== "string" ) {
```

Comment:
The default value of `processData` is true, so this change seems to be circumventing explicit user intention and I'm therefore opposed. It would also need unit tests before landing, but I'd first focus on the rationale. How did you encounter an issue in the first place?